qunicorn_core/core/transpiler/qrisp_transpiler.py

```diff
@@ -33,6 +35,51 @@ class QrispToQiskit(CircuitTranspiler, source="QRISP", target="QISKIT", cost=1):
 
     def transpile_circuit(self, circuit: Any) -> QiskitCircuit:
         assert isinstance(circuit, QrispCircuit)
-        converted = circuit.to_qiskit()
-        assert isinstance(converted, QiskitCircuit)
-        return converted
+        converted: QiskitCircuit = circuit.to_qiskit()
+        fused_bits = self._fuse_bits(converted)
+
+        assert isinstance(fused_bits, QiskitCircuit)
+        return fused_bits
+
+    @staticmethod
+    def _fuse_bits(circuit: QiskitCircuit) -> QiskitCircuit:
+        """
+        When converting a Qrisp circuit to Qiskit each qubit / clbit is in a separate register. This method fuses them
+        into one register.
+        @param circuit: Qiskit circuit with each qubit / clbit in a separate register
+        @return: a circuit where all qubits / clbits are in the same register
+        """
+        instructions: QuantumCircuitData = circuit.data
+        clbits_count = len(circuit.clbits)
+        qubits_count = len(circuit.qubits)
+        new_clreg = ClassicalRegister(clbits_count, name="clbits")
+        new_qreg = QuantumRegister(qubits_count, name="qubits")
+
+        clbit_map = {}
+        qubit_map = {}
+
+        for i, clbit in enumerate(circuit.clbits):
+            clbit_map[clbit._register.name] = i
+
+        for i, qubit in enumerate(circuit.qubits):
+            qubit_map[qubit._register.name] = i
+
+        inst: CircuitInstruction
+        new_instructions = []
+
+        for inst in instructions:
+            clbit_names = [clbit._register.name for clbit in inst.clbits]
```

Comment:
as a minor optimization: for temporary list comprehensions that get used only once one can use a generator comprehension (with round brackets) instead. This will not instantiate the entire list but create a lazy generator object instead. (definitely not required here but can be a nice trick to know if dealing with larger lists)